### PR TITLE
fix document building

### PR DESCRIPTION
**Pull request by aerorahul**
Change-Id: I7372755054e3cab2f86943e718f3e592b5f5396b
Use cmake provided FindLATEX.cmake

### CODE REVIEW

**Comment by aerorahul** [comment]
Note that `jcsda::develop`, lags `ecmwf::develop`.
`ecmwf::develop` has changed how it enables/disables features.
Some additional work will be required when `ecmwf::develop` is merged into `jcsda::develop`.

**Comment by mmiesch** [comment]
Thanks @aerorahul - latex is found for me but the documentation still isn't building.  Still, this is an improvement.

**Comment by aerorahul** [comment]
> Thanks @aerorahul - latex is found for me but the documentation still isn't building. Still, this is an improvement.

I think the COMPONENTS should be added `PDFLATEX`, `BIBTEX`, `MAKE_INDEX` and `HTLATEX`.
Those are used.
In addition `convert` program from `imagemagick` is used.

With all those found, I could build everything; pdf and html on my mac.

**Comment by mmiesch** [comment]
Thanks @aerorahul - I installed imagemagick and ford, I have all the other packages you mentioned, they are all found, and I'm building with  ```-DENABLE_DOCS=ON``` but I'm still not seeing anything useful in the atlas/doc directory.  The html directory is empty and there are no pdfs.

**Comment by aerorahul** [comment]
> Thanks @aerorahul - I installed imagemagick and ford, I have all the other packages you mentioned, they are all found, and I'm building with `-DENABLE_DOCS=ON` but I'm still not seeing anything useful in the atlas/doc directory. The html directory is empty and there are no pdfs.

The docs are not built with just `make -j<threads>`.  That only builds the code.
One has to explicitly do `make atlas_doc` or `make atlas_html` for the docs to be created.
Also, you cannot use make -j<threads> for creating docs.

**Comment by aerorahul** [comment]
> > Thanks @aerorahul - I installed imagemagick and ford, I have all the other packages you mentioned, they are all found, and I'm building with `-DENABLE_DOCS=ON` but I'm still not seeing anything useful in the atlas/doc directory. The html directory is empty and there are no pdfs.
> 
> The docs are not built with just `make -j<threads>`. That only builds the code.
> One has to explicitly do `make atlas_doc` or `make atlas_html` for the docs to be created.
> Also, you cannot use make -j for creating docs.

type `make help` in the  `atlas` directory`.  There are other ones too. see below.
```
... doc
... atlas_doc
... atlas_doc_html
... atlas_doc_pdf
... atlas_doc_code
... atlas-user-guide-pdf
... atlas-user-guide-html
```

**Comment by mmiesch** [comment]
Thanks but I'm still having problems - have you ever seen this?
~~~~~
miesch:~/JEDI/code/working_copy/build/atlas/doc/user-guide$ make atlas-user-guide-pdf
make[3]: *** [atlas/doc/user-guide/CMakeFiles/atlas-user-guide-pdf] Error 1
make[2]: *** [atlas/doc/user-guide/CMakeFiles/atlas-user-guide-pdf.dir/all] Error 2
make[1]: *** [atlas/doc/user-guide/CMakeFiles/atlas-user-guide-pdf.dir/rule] Error 2
make: *** [atlas/doc/user-guide/CMakeFiles/atlas-user-guide-pdf.dir/rule] Error 2
~~~~~

**Comment by aerorahul** [comment]
can you go to the top of the build directory and do the same command?
```
cd ~/JEDI/code/working_copy/build/atlas
make atlas-user-guide-pdf
```

**Comment by mmiesch** [comment]
There I get:
~~~~~
miesch:~/JEDI/code/working_copy/build/atlas$ make atlas-user-guide-pdf
make: *** No rule to make target `atlas-user-guide-pdf'.  Stop.
~~~~~
VERBOSE isn't much help.  Did you add any other options to ecbuild?

**Comment by aerorahul** [comment]
> There I get:
> 
> ```
> miesch:~/JEDI/code/working_copy/build/atlas$ make atlas-user-guide-pdf
> make: *** No rule to make target `atlas-user-guide-pdf'.  Stop.
> ```
> 
> VERBOSE isn't much help. Did you add any other options to ecbuild?

Make sure that you are actually running make at the top of the atlas build directory.
Is build your top level directory where you did ecbuild /path/to/atlas/clone?

Then there is where you should do make atlas-user-guide-pdf

do a make help at that level too

**Comment by mmiesch** [comment]
It is more informative if I build atlas on it's own - then I do see all the options at the top level - it looks like some cmake variable is empty:
~~~~~
$make VERBOSE=1  atlas-user-guide-pdf
[...]
cd /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas/doc/user-guide && export TEXINPUTS=/Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas//:/Users/miesch/JEDI/code/working_copy/build-atlas/doc/user-guide//: && --output-directory /Users/miesch/JEDI/code/working_copy/build-atlas/doc/user-guide /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas/doc/user-guide/user-guide.tex
/bin/sh: --output-directory: command not found
~~~~~

**Comment by mmiesch** [comment]
Yeah - it looks like PDFLATEX is not being set - I'll poke around a bit more.  Thanks for the help
~~~~
./user-guide/CMakeLists.txt:   ${PDFLATEX} --output-directory ${USERGUIDE} ${USERGUIDESRC}/user-guide.tex
~~~~~

**Comment by aerorahul** [comment]
> It is more informative if I build atlas on it's own - then I do see all the options at the top level - it looks like some cmake variable is empty:
> 
> ```
> $make VERBOSE=1  atlas-user-guide-pdf
> [...]
> cd /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas/doc/user-guide && export TEXINPUTS=/Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas//:/Users/miesch/JEDI/code/working_copy/build-atlas/doc/user-guide//: && --output-directory /Users/miesch/JEDI/code/working_copy/build-atlas/doc/user-guide /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/atlas/doc/user-guide/user-guide.tex
> /bin/sh: --output-directory: command not found
> ```

I did a clean clone and build
I have eckit and fckit already built and available as modules

```
git clone -b bugfix/docs github:jcsda/atlas
cd atlas
mkdir -p build; cd build
ecbuild -DENABLE_DOCS=ON ..
make -j4
make doc
ls doc/user-guide
CMakeFiles           core-functionalities user-guide.idx       user-guide.toc
CTestTestfile.cmake  getting-started      user-guide.log       version.tex
Makefile             html                 user-guide.out
cmake_install.cmake  user-guide.aux       user-guide.pdf

```

**Comment by mmiesch** [comment]
Ok - I found it - it looks like the old ```FindLatex``` in the develop branch (which works for me) defined a ```PDFLATEX``` variable that my cmake 3.17 ```FindLATEX``` does not.  Instead, it defines ```PDFLATEX_COMPILER```.  So, with this substitution the atlas-user-guide-pdf target works for me.  I suspect similar replacements will be needed for the other components

**Comment by aerorahul** [comment]
> Yeah - it looks like PDFLATEX is not being set - I'll poke around a bit more. Thanks for the help
> 
> ```
> ./user-guide/CMakeLists.txt:   ${PDFLATEX} --output-directory ${USERGUIDE} ${USERGUIDESRC}/user-guide.tex
> ```



> Ok - I found it - it looks like the old `FindLatex` in the develop branch (which works for me) defined a `PDFLATEX` variable that my cmake 3.17 `FindLATEX` does not. Instead, it defines `PDFLATEX_COMPILER`. So, with this substitution the atlas-user-guide-pdf target works for me. I suspect similar replacements will be needed for the other components

Apologies @mmiesch.
It looks like PDFLATEX_COMPILER is the one being set. So we should use that.

File: doc/user-guide/CMakeLists.txt [hunk 1]
```diff
@@ -69,8 +69,8 @@ ENDFOREACH()
 # PDF GENERATION ==============================================================
 add_custom_target(atlas-user-guide-pdf
    export TEXINPUTS=${CMAKE_SOURCE_DIR}//:${CMAKE_CURRENT_BINARY_DIR}//: &&
-   ${PDFLATEX} --output-directory ${USERGUIDE} ${USERGUIDESRC}/user-guide.tex
+   ${PDFLATEX_COMPILER} --output-directory ${USERGUIDE} ${USERGUIDESRC}/user-guide.tex
```
Comment (aerorahul) [inline doc/user-guide/CMakeLists.txt:72]: This is not a legitimate return of FindLATEX.cmake.
The FindLATEX.cmake returns PDFLATEX.
Please revert this change.

**Comment by mmiesch** [comment]
The package name isn't always the same as the executable, is it?  The way atlas is using it is as an executable program contained in a variable.  Isn't that what ```PDFLATEX_COMPILER``` is?  I'm not replacing the ```find_package``` command - just where the executable is called.  If you look at the FindLATEX, it's using find_program to define ```PDFLATEX_COMPILER```
~~~~~
miesch:/usr/local/share/cmake/Modules$ grep PDFLATEX FindLATEX.cmake 
  PDFLATEX_COMPILER:    path to the PdfLaTeX compiler
  PDFLATEX
  find_package(LATEX COMPONENTS PDFLATEX)
find_program(PDFLATEX_COMPILER
if (PDFLATEX_COMPILER)
  set(LATEX_PDFLATEX_FOUND TRUE)
  set(LATEX_PDFLATEX_FOUND FALSE)
  PDFLATEX_COMPILER
~~~~~

**Comment by aerorahul** [comment]
> The package name isn't always the same as the executable, is it? The way atlas is using it is as an executable program contained in a variable. Isn't that what `PDFLATEX_COMPILER` is? I'm not replacing the `find_package` command - just where the executable is called. If you look at the FindLATEX, it's using find_program to define `PDFLATEX_COMPILER`
> 
> ```
> miesch:/usr/local/share/cmake/Modules$ grep PDFLATEX FindLATEX.cmake 
>   PDFLATEX_COMPILER:    path to the PdfLaTeX compiler
>   PDFLATEX
>   find_package(LATEX COMPONENTS PDFLATEX)
> find_program(PDFLATEX_COMPILER
> if (PDFLATEX_COMPILER)
>   set(LATEX_PDFLATEX_FOUND TRUE)
>   set(LATEX_PDFLATEX_FOUND FALSE)
>   PDFLATEX_COMPILER
> ```

You are right.  Similar edits will be necessary for the other components I think.

**Comment by mmiesch** [comment]
Good, I'm glad you agree - it seems to work.  I think the only other thing that needs to be changed is a call to HTLATEX.  While testing I ran into a bug in my latex so I'm installing mactex now - taking forever.  I'll push the change soon.

**Comment by aerorahul** [comment]
> Good, I'm glad you agree - it seems to work. I think the only other thing that needs to be changed is a call to HTLATEX. While testing I ran into a bug in my latex so I'm installing mactex now - taking forever. I'll push the change soon.

I brew-ed mactex.

**Comment by mmiesch** [comment]
Ok, all the various pdf, html, and code doc build targets are working for me now.  Thanks @aerorahul for your help

**Review by mmiesch — APPROVED** [review]

**Review by ytremolet — CHANGES_REQUESTED** [review]
We should issue PRs into ECMWF's develop, not JCSDA's.

**Comment by aerorahul** [comment]
@mmiesch @ytremolet 
Is it permitted to sync `jcsda::atlas/develop` with `ecmwf::atlas/develop`?
If that is done, this branch is redundant and PR can be issued to ECMWF

**Comment by aerorahul** [comment]
I opened a PR on the ECMWF::atlas/develop [here](https://github.com/ecmwf/atlas/pull/18). I cannot add reviewers, so I mentioned @ytremolet in the PR.

**Comment by mmiesch** [comment]
Thanks @aerorahul

**Comment by ytremolet** [comment]
We should bring our develop to be up to date with ecmwf's. In fact I would really like if we could make that automatic so it's always up to date.

**Comment by aerorahul** [comment]
@mmiesch @ytremolet 
I am closing this PR.
A PR on ECMWF::atlas/develop [here](https://github.com/ecmwf/atlas/pull/18) was accepted and closed.  We should sync up with ECMWF.

**Pull request closed without merging**